### ISSUE TITLE
Add option to rotate the calibration canvas

### ISSUE THREAD

**Issue by courtneypattison** [Pattern-Projector/pattern-projector#44]
Some people have setups that are flipped from their computer. Someone asked to have an option to rotate the canvas to make calibration more intuitive.

**Comment by hluedeke** [comment]
Will this even be possible? Like, how are they going to use a mouse to find the correct corner? If they use keys, then shouldn't the color-coded corners solve the problem?

...I'm having trouble visualizing this one technically

**Comment by courtneypattison** [comment]
I think you're right! The only way it would make sense is by remapping arrow keys but then I'd be scared someone else would stumble into it and be confused by what's happening. Maybe close this and suggest changing the display settings on their desktop to make it more intuitive.

**Comment by hluedeke** [comment]
> Maybe close this and suggest changing the display settings on their desktop to make it more intuitive.

That, or maybe we try to get them to use colored corners and see if they can figure it out from there. I'm not even sure if desktops have a setting like that 😂

Do they not have this problem in projector programs (Adobe, Inkscape, etc) already today?

**Comment by hluedeke** [comment]
Hmmm, maybe during calibration...they can "swap" the colored corners of the grid when the corner highlights are on? This will of course only affect arrow keys...but maybe?

And then keep that transform for PDFs? Maybe that would work...

**Comment by courtneypattison** [comment]
<img width="823" alt="rotate_display" src="https://github.com/Pattern-Projector/pattern-projector/assets/5465100/d3646848-54a3-4cc0-8d21-4a19f6ed600f">
On a mac you can rotate the display to match your setup so I think we should just leave it at that. It seems like a pretty special case! I guess we could return to this if it keeps coming up.

**Comment by courtneypattison** [comment]
Yes! It seems like people so far prefer keyboard based movement.

**Comment by hluedeke** [comment]
Ok, if that's the case, then I can work on this if you want. I actually think I get it now 😂

**Comment by hluedeke** [comment]
You'll notice I opened a PR for key management and not for rotation. I started to do rotation, and it turned out to be a massive pain in the rear. Toggling the corners wasn't too bad - it was "flipping" the calibration grid while not actually flipping the calibration grid.

My plan was not to apply css rotate, but to modify the height/width parameter on the grid itself by swapping them.
I tried swapping height and width of the grid before calling ```draw()```, but that turned out to only partially work. Some of the grid rotated, but the black background did not (and honestly, maybe some of the other parts of the grid didn't either, it was hard to tell and pretty wonky).

After that, I decided to focus on the precursor to rotation instead, the key handlers.
